modules/features2d/test/test_matchers_algorithmic.cpp

```diff
@@ -558,4 +558,41 @@ TEST( Features2d_DMatch, read_write )
     ASSERT_NE( strstr(str.c_str(), "4.5"), (char*)0 );
 }
 
+TEST( Features2d_FlannBasedMatcher, read_write )
+{
+    static const char* ymlfile = "%YAML:1.0\n---\n"
+    "format: 3\n"
+    "indexParams:\n"
+    "   -\n"
+    "      name: algorithm\n"
+    "      type: 9\n"  // FLANN_INDEX_TYPE_ALGORITHM
+    "      value: 6\n"// this line is changed!
```

Comment:
I basically copied 1:1 one of java tests that, since I did not want to debug java tests. If we remove FLANN sometimes (not in OpenCV 4.0, I guess), this test can be removed as well.